### PR TITLE
Givemethiscase

### PR DESCRIPTION
**Pull request by dlarge3**

### CODE REVIEW

File: src/test/java/com/hocs/test/pages/markup/MarkUpDecision.java [hunk 1]
```diff
@@ -142,4 +142,6 @@ public void assertTopicIsRequiredErrorMessage() {
         assertThat(topicIsRequiredErrorMessage.getText(), is("Topic is required"));
     }
 
+
```
Comment (kylesmithten10) [inline src/test/java/com/hocs/test/pages/markup/MarkUpDecision.java]: can remove these new lines

File: src/test/java/com/hocs/test/pages/markup/MarkupFull.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    Topics topics;
+
+    MarkUpDecision markUpDecision;
+
+    public void moveCaseFromMarkupToInitialDraft() {
+        markUpDecision.clickPolicyResponseRadioButton();
+        clickContinueButton();
+        topics.clickAddTopicButton();
+        topics.enterRealTopic();
+        sleep(1000);
```
Comment (kylesmithten10) [inline src/test/java/com/hocs/test/pages/markup/MarkupFull.java]: waitABit(1000);

File: src/test/java/com/hocs/test/glue/DataInputStepDefs.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void iEnterCorrespondenceDataManually() {
         genericInputStepDefs.fillMandatoryFields("Data Input");
         dataInput.clickContinueButton();
         dataInput.clickAddCorrespondentLink();
-        recordCorrespondentDetails.fillMandatoryFields();
+        recordCorrespondentDetails.fillMandatoryCorrespondentFields();
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/DataInputStepDefs.java]: Data Input is 2 pages
Correspondence Details
Correspondent Details
So I have updated the `fillAllMandatoryFields` to `fillAllMandatoryCorrespondenceFields`

File: src/test/java/com/hocs/test/glue/DataInputStepDefs.java [hunk 4]
```diff
@@ -170,35 +176,19 @@ public void clickContinueButton() {
 
     @When("^I click the continue button at the data input stage$")
     public void userDoesNotEnterDateCorrespondenceWasSentDataInputStage() {
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/DataInputStepDefs.java]: removed method

File: src/test/java/com/hocs/test/glue/DataInputStepDefs.java [hunk 5]
```diff
@@ -170,35 +176,19 @@ public void clickContinueButton() {
...
-        dataInput.enterDayOfCorrespondenceSent("01");
-        dataInput.enterMonthOfCorrespondenceSent("01");
-        dataInput.enterYearOfCorrespondenceSent("2019");
-        dataInput.clickEmailCorrespondenceChannelRadioButton();
-        dataInput.clickContinueButton();
         dataInput.clickFinishButton();
     }
 
     @When("^I click the continue button on the is the correspondent an MP screen$")
     public void userDoesNotSelectPrimaryCorrespondentTypeRadioButton() {
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/DataInputStepDefs.java]: removed method

File: src/test/java/com/hocs/test/glue/DataInputStepDefs.java [hunk 6]
```diff
@@ -170,35 +176,19 @@ public void clickContinueButton() {
...
-        dataInput.clickEmailCorrespondenceChannelRadioButton();
-        dataInput.selectN10ResponseYesRadioButton();
-        dataInput.clickContinueButton();
-        dataInput.sleep(500);
         dataInput.clickAddCorrespondentLink();
         dataInput.clickContinueButton();
-
     }
 
     @When("^I click the add button on the add member of parliament screen$")
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/DataInputStepDefs.java]: allocateCaseAndSetDatesAndSelectNoForCopyToNumberTenAndSayTheCorrespondentIsAMemberAndFinallyClickTheAddButton();

File: src/test/java/com/hocs/test/glue/GenericInputStepDefs.java [hunk 7]
```diff
@@ -114,7 +114,15 @@ public void clickLink(String name) {
     @When("^I enter \"([^\"]*)\" in the \"([^\"]*)\" field")
     public void iEnterTextIntoTheNominatedField(String input, String element) {
         switch (element.toUpperCase()) {
-            case "":
+            case "ADD A MEMBER OF PARLIAMENT":
+                setSessionVariable("fullName").to(input);
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]:
```
    @When("^I enter \"([^\"]*)\" in the \"([^\"]*)\" field")
    public void iEnterTextIntoTheNominatedField(String input, String element) {
        switch (element.toUpperCase()) {
            case "ADD A MEMBER OF PARLIAMENT":
                setSessionVariable("fullName").to(input);
                recordCorrespondentDetails.addAMemberOfParliamentCorrespondent(input);
                break;
            case "FULL NAME" :
                setSessionVariable("fullName").to(input);
                break;
            case "ADDRESS" :
                break;
            default:
                pendingStep(element + " is not defined within " + getMethodName());
        }
    }
```
Dont see this

[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]: To assert later

File: src/test/java/com/hocs/test/glue/GenericInputStepDefs.java [hunk 8]
```diff
@@ -114,7 +114,15 @@ public void clickLink(String name) {
...
+                recordCorrespondentDetails.addAMemberOfParliamentCorrespondent(input);
+                break;
+            case "FULL NAME" :
+                setSessionVariable("fullName").to(input);
+
+                break;
+            case "ADDRESS" :
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]: I will need this.

File: src/test/java/com/hocs/test/glue/GenericInputStepDefs.java [hunk 9]
```diff
@@ -126,6 +134,11 @@ public void fillMandatoryFields(String pageName) {
         switch (pageName.toUpperCase()) {
             case "DATA INPUT":
                 dataInput.fillAllMandatoryFields();
+                dataInput.clickContinueButton();
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]: required or next step breaks

File: src/test/java/com/hocs/test/glue/GenericInputStepDefs.java [hunk 10]
```diff
@@ -126,6 +134,11 @@ public void fillMandatoryFields(String pageName) {
...
+                break;
+            case "CORRESPONDENT DETAILS" :
+                recordCorrespondentDetails.fillMandatoryCorrespondentFields();
+                dataInput.clickAddButton();
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]: required or next step breaks

File: src/test/java/com/hocs/test/glue/GenericInputStepDefs.java [hunk 11]
```diff
@@ -126,6 +134,11 @@ public void fillMandatoryFields(String pageName) {
         switch (pageName.toUpperCase()) {
             case "DATA INPUT":
                 dataInput.fillAllMandatoryFields();
+                dataInput.clickContinueButton();
+                break;
+            case "CORRESPONDENT DETAILS" :
+                recordCorrespondentDetails.fillMandatoryCorrespondentFields();
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/GenericInputStepDefs.java]: The step def is called 'I fill all mandatory fields on the """""" page' ?!

File: src/test/java/com/hocs/test/glue/NavigationStepDefs.java [hunk 12]
```diff
@@ -122,6 +125,11 @@ public void navigateToTeamPage(String teamPage) {
 
     }
 
+    @When("^I get a \"([^\"]*)\" case at \"([^\"]*)\" stage$")
+    public void reallyGiveMeACase(String caseType, String stage) {
+        fetch.giveMeACase(caseType, stage);
+    }
+
     @When("^I click the back to dashboard button$")
     public void clickBackToDashboardButtonOnAllocateCasePage() {
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/glue/NavigationStepDefs.java]: Other way around but agree

File: src/test/java/com/hocs/test/pages/Page.java [hunk 13]
```diff
@@ -355,6 +358,10 @@ public void clickRejectButton() {
         rejectButton.click();
     }
 
+    public void clickLogoutButton() {
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/Page.java]: need to do them all so task for a refactoring day imo

File: src/test/java/com/hocs/test/pages/data_input/RecordCorrespondentDetails.java [hunk 14]
```diff
@@ -131,15 +135,26 @@ public String getPrimaryCorrespondent() {
         return primaryCorrespondentName.getText();
     }
 
+    public void addAMemberOfParliamentCorrespondent(String member) {
+        memberDropdownInitialize.click();
+        //memberDropdown.sendKeys(member);
+        typeInto(memberDropdown, member);
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/data_input/RecordCorrespondentDetails.java]: You need to click into the dropdown first but ours is a reactJS typeahead dropdown so the actual input element doesnt exist until you have activated it hence `memberDropdownInitialize`

[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/data_input/RecordCorrespondentDetails.java]: May not be the same for you

File: src/test/java/com/hocs/test/pages/markup/MarkupFull.java (hunk 2 above)
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/markup/MarkupFull.java]: As clickOn();

File: src/test/java/com/hocs/test/pages/data_input/DataInput.java [hunk 15]
```diff
@@ -140,14 +140,23 @@ public void tickSendCopyToNumber10() {
...
         successfulCaseCreation.selectCaseReferenceNumberViaXpath();
         workstacks.clickAllocateToMeButton();
-        fillAllMandatoryFields();
+        fillAllMandatoryCorrespondenceFields();
+        clickContinueButton();
+        recordCorrespondentDetails.addAMemberOfPublicCorrespondent();
+        clickFinishButton();
+    }
+
+    public void dataInputFullFlowWCopyToN10() {
```
Comment (kylesmithten10) [inline src/test/java/com/hocs/test/pages/data_input/DataInput.java]: dataInputFullFlowWCopyToN10
typo?
dataInputFullFlowCopyToN10

File: src/test/java/com/hocs/test/pages/data_input/DataInput.java [hunk 16]
```diff
@@ -206,15 +215,52 @@ public void enterReferenceText() {
...
         enterMonthOfCorrespondenceSent(todayPlusNDaysGetMonth(-2));
         enterYearOfCorrespondenceSent(todayPlusNDaysGetYear(-2));
         enterDayOfCorrespondenceReceived(getCurrentDay());
         enterMonthOfCorrespondenceReceived(getCurrentMonth());
         enterYearOfCorrespondenceReceived(getCurrentYear());
         emailOriginalChannelRadioButton.click();
+        selectN10ResponseYesRadioButton();
+    }
+
+    public void getToAddMemberOfParliamentPrerequisites() {
```
Comment (kylesmithten10) [inline src/test/java/com/hocs/test/pages/data_input/DataInput.java]: getToAddMemberOfParliamentPrerequisites and getToRecordCorrespondentDetailsPrerequisites seem to do the same thing ?

File: src/test/java/com/hocs/test/pages/data_input/RecordCorrespondentDetails.java (hunk 14 above)
[the remark this answers is not in the document]
Comment (kylesmithten10) [reply src/test/java/com/hocs/test/pages/data_input/RecordCorrespondentDetails.java]: thanks for the info, if it works for you then thats fine

File: src/test/resources/features/EndToEnd.feature [hunk 17]
```diff
@@ -91,7 +91,18 @@ Feature: HOCS is able to move cases through the entire flow
    #   | DCU TRO |
    #   | DCU N10 |
 
-  #Remember to do Copy to #10
+  @EndToEnd @DCUMIN @Critical
+  Scenario: End to End flow and send a copy to N10
```
Comment (kylesmithten10) [inline src/test/resources/features/EndToEnd.feature]: ideally should only have 1 when and 1 then, if you have a lot of setup just have it as part of your given using * or and
eg:
```java
Scenario: Unlock Document
    Given test
    And test 
    And test
    When test 
    Then test
```
or
```java
Scenario: Unlock Document
    Given test
    * test
    * test
    When test
    Then test
```

File: src/test/java/com/hocs/test/pages/data_input/DataInput.java (hunk 16 above)
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/data_input/DataInput.java]: Record Correspondent Details - Member of the public
Add Member of Parliament - Member of Parliament

File: src/test/java/com/hocs/test/pages/data_input/DataInput.java (hunk 15 above)
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/data_input/DataInput.java]: W = With

File: src/test/java/com/hocs/test/pages/Page.java [hunk 18]
```diff
@@ -395,6 +395,24 @@ public void clickEmailFeedbackButton() {
         emailFeedbackButton.click();
     }
 
+    public void clickTheRequiredButton(String buttonName) {
```
[the remark this answers is not in the document]
Comment (dlarge3) [reply src/test/java/com/hocs/test/pages/Page.java]: We have a clickTheButton as a stepDef so updated to clickThisButton

**Pull request merged**